### PR TITLE
Exempt typing.overload from duplicate detection

### LINKED ISSUE #320: F811 (redefinition of unused variable) incorrectly triggered with @overload decorator
PEP 484 [introduced](https://www.python.org/dev/peps/pep-0484/#function-method-overloading) the `@overload` decorator for type annotations.

> [...] a series of `@overload`-decorated definitions must be followed by exactly one non-@overload-decorated definition (for the same function/method). The `@overload`-decorated definitions are for the benefit of the type checker only, since they will be overwritten by the non-`@overload`-decorated definition, while the latter is used at runtime but should be ignored by a type checker.

The following should pass PyFlakes without errors, but currently triggers F811 errors:

```python
from typing import overload


@overload
def utf8(value: None) -> None:
    pass


@overload
def utf8(value: bytes) -> bytes:
    pass


@overload
def utf8(value: str) -> bytes:
    pass


def utf8(value):
    pass  # actual implementation
```
```
test.py:9:1: F811 redefinition of unused 'utf8' from line 4
test.py:14:1: F811 redefinition of unused 'utf8' from line 9
test.py:19:1: F811 redefinition of unused 'utf8' from line 14
```

### PR DESCRIPTION
**Pull request by asottile**
Resolves #320

### CODE REVIEW

File: pyflakes/checker.py [hunk 1]
```diff
@@ -477,6 +477,28 @@ def getNodeName(node):
         return node.name
 
 
+def isOverload(value, scope):
```
[the remark this answers is not in the document]
Comment (asottile) [reply pyflakes/checker.py]: the existing code is camel case, wouldn't this be inconsistent? I was attempting to match the existing code

**Comment by asottile** [comment]
(your example of what I _should_ use also is invalid, your first function has parameter 1 typed as both `str` and `None`)

**Comment by jayvdb** [comment]
Ah sorry about that `str`/`None`, I was melding the pep example with your test function names, and messed up.

File: pyflakes/checker.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jayvdb) [reply pyflakes/checker.py]: Ya, understood, but new code getting merged is typically pep 8 compliant these days.  Lots of the non compliant code is very old, and change for changes sake is quite harmful.

**Comment by asottile** [comment]
@jayvdb what did you want me to change about the test? please clarify

**Comment by jayvdb** [comment]
The [PEP](https://www.python.org/dev/peps/pep-0484/#function-method-overloading) uses
```
from typing import overload

class bytes:
    ...
    @overload
    def __getitem__(self, i: int) -> int: ...
    @overload
    def __getitem__(self, s: slice) -> bytes: ...
```

It is because of the inputs and outputs that this is beneficial and can not be expressed using `typing.Union`.

I thought it would be useful to include that, but it isnt necessary.

**Comment by asottile** [comment]
> The [PEP](https://www.python.org/dev/peps/pep-0484/#function-method-overloading) uses
> 
> ```
> from typing import overload
> 
> class bytes:
>     ...
>     @overload
>     def __getitem__(self, i: int) -> int: ...
>     @overload
>     def __getitem__(self, s: slice) -> bytes: ...
> ```
> 
> It is because of the inputs and outputs that this is beneficial and can not be expressed using `typing.Union`.
> 
> I thought it would be useful to include that, but it isnt necessary.

The minimal example in the test also cannot be expressed via union:

```
None -> None
int -> int
```

**Comment by asottile** [comment]
my simple example could however be expressed via `TypeVar` -- but I really don't think that matters -- the point isn't that it's meaningful, it's that it uses `@override` correctly.

**Comment by jayvdb** [comment]
Yes, but it is only using `typing` and not combined with Python 3 function annotations.  I was wondering if that might be another possible source of problems related to overload inside pyflakes, and if it was being avoided due to other bugs.  Apologies for not being more clear.

Anyways, I have now tested it for myself and it works ok, at least with this basic test.
```py
        @typing.overload
        def h(s : None) -> None:
            pass

        @typing.overload
        def h(s : int) -> int:
            pass

        def h(s):
            return s
```

**Comment by asottile** [comment]
I avoided the python3 annotations because I didn't want to have to disable the test for python2.x

**Comment by jayvdb** [comment]
Much appreciated.  This is a very important fix IMO.

**Pull request merged**